### PR TITLE
Seacas may2020 snapshot

### PR DESCRIPTION
**Pull request by gsjaardema**
@trilinos/seacas

### CODE REVIEW

**Comment by gsjaardema** [comment]
@bartlettroscoe According to the documentation for TriBITS, `C99` mode is supposed to be the default, but based on the error message for the intel build, it looks like it is using `C89` mode since it does not recognized the variable declaration inside the `for` statement.  This is even with my SEACAS CMakeLists.txt explicitly requesting `C99` mode. 

Any ideas what I am doing wrong?

**Comment by gsjaardema** [comment]
This should be ready for review now.  Sorry for the amount of changes; many of them are just some float->dble and ifix->int changes in fortran.  Tried to split out into multiple commits, but not sure that makes it any easier.

**Comment by gsjaardema** [comment]
@bartlettroscoe @prwolfe 

Any chance of getting this merged?  Sorry for the size, but parts of it contain some optimizations that would be helpful for milestone runs...

**Comment by prwolfe** [comment]
I am looking through it.

File: packages/seacas/libraries/exodus/CMakeLists.txt [hunk 1]
```diff
@@ -38,13 +38,16 @@ TRIBITS_ADD_LIBRARY(
   SOURCES ${SOURCES} ${DEP_SOURCES}
 )
 
+set_property(TARGET exodus PROPERTY C_STANDARD 99)
```
[the remark this answers is not in the document]
Comment (gsjaardema) [reply packages/seacas/libraries/exodus/CMakeLists.txt:41]: It is only needed for the library itself I think. should not affect down-stream customers?

File: packages/seacas/libraries/ioss/src/cgns/CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,18 @@
...
 
+IF (TPL_ENABLE_MPI)
+ASSERT_DEFINED(${PROJECT_NAME}_ENABLE_Zoltan)
+IF (NOT ${PROJECT_NAME}_ENABLE_Zoltan)
+  ADD_DEFINITIONS(-DNO_ZOLTAN_SUPPORT)
+ENDIF()
+
+IF (${PROJECT_NAME}_ENABLE_ParMETIS)
+ELSE()
+  ADD_DEFINITIONS(-DNO_PARMETIS_SUPPORT)
```
[the remark this answers is not in the document]
Comment (gsjaardema) [reply packages/seacas/libraries/ioss/src/cgns/CMakeLists.txt:13]: The definition is only needed for the build of `Iocgns` library.  Does not affect the client/customer.

File: packages/seacas/libraries/ioss/src/cgns/CMakeLists.txt [hunk 3]
```diff
@@ -2,6 +2,18 @@
 SET(HEADERS "")
 SET(SOURCES "")
 
+IF (TPL_ENABLE_MPI)
+ASSERT_DEFINED(${PROJECT_NAME}_ENABLE_Zoltan)
+IF (NOT ${PROJECT_NAME}_ENABLE_Zoltan)
+  ADD_DEFINITIONS(-DNO_ZOLTAN_SUPPORT)
```
[the remark this answers is not in the document]
Comment (gsjaardema) [reply packages/seacas/libraries/ioss/src/cgns/CMakeLists.txt:8]: Just locally built code.

**Comment by gsjaardema** [comment]
@prwolfe Yes sorry for the size.  "it looks decent at least" -- The best compliment I've had in awhile.  ;-)

**Comment by prwolfe** [comment]
It’s something I struggle to get in my code – just saying.

**Pull request merged**